pkg/provision/provision.go

```diff
@@ -58,4 +57,21 @@ var (
 
 // ErrTemporary is return when a reservation source failed to contact the BCDB
 // user usually want to retry after getting this error
-var ErrTemporary = errors.New("network error contacting BCDB")
+type ErrTemporary struct {
+	err error
+}
+
+// NewErrTemporary wrap an error and mark it as temporary
+func NewErrTemporary(err error) error {
+	return ErrTemporary{err: err}
+}
+
+// Error implements the errors.Error interface
+func (e ErrTemporary) Error() string {
+	return e.Error()
```

Comment:
This will cause a stack overflow.

Reply:
wow nice catch, should be `e.err.Error()`.